src/wasm/kotlin/tests/arcs/ParticleApiTest.kt

```diff
@@ -0,0 +1,145 @@
+package arcs
+
+import kotlin.native.internal.ExportForCppRuntime
+
+class HandleSyncUpdateTest : Particle() {
+    private val input1 = Singleton { Test_Data() }
+    private val input2 = Singleton { Test_Data() }
+    private val output = Collection { Test_Data() }
+
+    init {
+        registerHandle("input1", input1)
+        registerHandle("input2", input2)
+        registerHandle("output", output)
+    }
+
+    override fun onHandleSync(handle: Handle, allSynced: Boolean) {
+        val out = Test_Data(txt = "sync:${handle.name}", flg = allSynced)
+        output.store(out)
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        val input = (handle as Singleton<*>).get() as Test_Data?
+        val out = input?.let { Test_Data(input.num, "update:${handle.name}") }
+                ?: Test_Data(txt = "unexpected handle name: ${handle.name}")
+
+        output.store(out)
+    }
+}
+
+@Retain
+@ExportForCppRuntime("_newHandleSyncUpdateTest")
+fun constructHandleSyncUpdateTest(): WasmAddress = HandleSyncUpdateTest().toWasmAddress()
+
+
+class RenderTest : Particle() {
+    private val flags = Singleton { Test_RenderFlags() }
+
+    init {
+        registerHandle("flags", flags)
+    }
+
+    override fun getTemplate(slotName: String): String {
+        return "abc"
+    }
+
+    override fun populateModel(slotName: String, model: Map<String, String>): Map<String, String> {
+        return mapOf("foo" to "bar")
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        val flags = flags.get()
+        flags?.let {
```

Comment:
this could be combined with the previous line, and the local `flags` val could be removed